tests/sentry/integrations/msteams/test_link_identity.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import absolute_import
+
+import responses
+import time
+import json
+
+from sentry.utils.compat.mock import patch
+
+from sentry.models import (
+    Identity,
+    IdentityProvider,
+    IdentityStatus,
+    Integration,
+    OrganizationIntegration,
+)
+from sentry.testutils import TestCase
+from sentry.integrations.msteams.link_identity import build_linking_url
+
+
+class MsTeamsIntegrationLinkIdentityTest(TestCase):
+    def setUp(self):
+        super(TestCase, self).setUp()
+        self.user1 = self.create_user(is_superuser=False)
+        self.user2 = self.create_user(is_superuser=False)
+        self.org = self.create_organization(owner=None)
+        self.team = self.create_team(organization=self.org, members=[self.user1, self.user2])
+
+        self.login_as(self.user1)
+
+        self.integration = Integration.objects.create(
+            provider="msteams",
+            name="Hogwarts",
+            external_id="1_50l3mnly_5w34r",
+            metadata={
+                "service_url": "https://smba.trafficmanager.net/amer",
+                "access_token": "3ld3rw4nd",
+                "expires_at": int(time.time()) + 86400,
+            },
+        )
+        OrganizationIntegration.objects.create(organization=self.org, integration=self.integration)
+
+        self.idp = IdentityProvider.objects.create(
+            type="msteams", external_id="1_50l3mnly_5w34r", config={}
+        )
+
+    @responses.activate
+    @patch("sentry.integrations.msteams.link_identity.unsign")
+    def test_basic_flow(self, unsign):
+        unsign.return_value = {
+            "integration_id": self.integration.id,
+            "organization_id": self.org.id,
+            "teams_user_id": "a_p_w_b_d",
+            "team_id": "1_50l3mnly_5w34r",
+            "tenant_id": "h0g5m34d3",
+        }
+
+        linking_url = build_linking_url(
+            self.integration, self.org, "a_p_w_b_d", "1_50l3mnly_5w34r", "h0g5m34d3",
+        )
+
+        resp = self.client.get(linking_url)
+
+        assert resp.status_code == 200
+        self.assertTemplateUsed(resp, "sentry/auth-link-identity.html")
+
+        def user_conversation_id_callback(request):
+            payload = json.loads(request.body)
+            if payload["members"] == [{"id": "a_p_w_b_d"}] and payload["channelData"] == {
```

Comment:
@KKaempen What happens in the else condition? We should still return something, like an error status code.

Reply:
Would something like `(404, {}, json.dumps({}))` be OK?

Reply:
I think so